SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/CrossServerAbilityActivationTest/GA_IncrementSpyValue.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "GA_IncrementSpyValue.h"
+#include "GameplayTagContainer.h"
+#include "SpyValueGASTestActor.h"
+
+UGA_IncrementSpyValue::UGA_IncrementSpyValue()
+{
+	InstancingPolicy = EGameplayAbilityInstancingPolicy::NonInstanced;
+	NetExecutionPolicy = EGameplayAbilityNetExecutionPolicy::ServerOnly;
+
+	AbilityTags.AddTag(UGA_IncrementSpyValue::GetTriggerTag());
+
+	FAbilityTriggerData Trigger;
+	Trigger.TriggerSource = EGameplayAbilityTriggerSource::GameplayEvent;
+	Trigger.TriggerTag = UGA_IncrementSpyValue::GetTriggerTag();
+	AbilityTriggers.Add(Trigger);
+}
+
+void UGA_IncrementSpyValue::ActivateAbility(const FGameplayAbilitySpecHandle Handle, const FGameplayAbilityActorInfo* ActorInfo,
+											const FGameplayAbilityActivationInfo ActivationInfo, const FGameplayEventData* TriggerEventData)
+{
+	if (HasAuthorityOrPredictionKey(ActorInfo, &ActivationInfo))
+	{
+		if (!CommitAbility(Handle, ActorInfo, ActivationInfo))
+		{
+			EndAbility(CurrentSpecHandle, CurrentActorInfo, CurrentActivationInfo, true, true);
+			return;
+		}
+
+		ASpyValueGASTestActor* TestActor = static_cast<ASpyValueGASTestActor*>(ActorInfo->OwnerActor.Get());
+		if (TestActor == nullptr)
```

Comment:
Just want to clarify intent: do you want to check the type of `OwnerActor`? `static_cast` does no such checks, so `TestActor == nullptr` is essentially equivalent to `ActorInfo->OwnerActor.IsValid()`

Reply:
Yep that was the intent, I misunderstood about `static_cast` - thanks for clarifying!